ambry-api/src/main/java/com/github/ambry/network/RequestInfo.java

```diff
@@ -31,6 +32,8 @@ public class RequestInfo {
   private long requestEnqueueTime = -1;
   private long requestSendTime = -1;
   private long responseHeaderReceiveTime = -1;
+  private final AtomicLong networkTimeOutMs = new AtomicLong(0);
+  private final AtomicLong finalTimeOutMs = new AtomicLong(0);
```

Comment:
Nit: these aren't timeout values. These are timeTaken/timeSpent.

Reply:
+1

Reply:
Hmm, these are actually timeout values for this request. Just that they are incremented at run time on top of configured value.